config/smarty.config.inc.php

```diff
@@ -177,9 +177,7 @@ function smartyCleanHtml($data)
 function smartyClassname($classname)
 {
     $classname = Tools::replaceAccentedChars(strtolower($classname));
-    $classname = preg_replace('/[^A-Za-z0-9]/', '-', $classname);
-    $classname = preg_replace('/[-]+/', '-', $classname);
-
+    $classname = preg_replace(['/[^A-Za-z0-9-_]/', '/-{3,}/', '/-+$/'], ['-', '-', ''] , $classname);
```

Comment:
```suggestion
    $classname = preg_replace(['/[^A-Za-z0-9-_]/', '/-{3,}/', '/-+$/'], ['-', '-', ''] , $classname);
```

Just notice something,n maybe we can use trim or ltrim or trim instead of preg_replace the third element, wdyt? Do we need to remove dash at the end only or also at the beginning?

Reply:
I have already crossed syntax of the type `--classname` or `-classname`. I've never seen the use of a hyphen at the end. 
So personally I would just leave the final cleanup.